modules/bootstrapped/src/generated/smithy4s/example/DefaultVariants.scala

```diff
@@ -0,0 +1,25 @@
+package smithy4s.example
+
+import smithy4s.Hints
+import smithy4s.Schema
+import smithy4s.ShapeId
+import smithy4s.ShapeTag
+import smithy4s.schema.Schema.string
+import smithy4s.schema.Schema.struct
+
+final case class DefaultVariants(req: String, reqDef: String = "default", optDef: String = "default", opt: Option[String] = None)
+
+object DefaultVariants extends ShapeTag.Companion[DefaultVariants] {
+  val id: ShapeId = ShapeId("smithy4s.example", "DefaultVariants")
+
+  val hints: Hints = Hints.empty
+
+  implicit val schema: Schema[DefaultVariants] = struct(
+    string.required[DefaultVariants]("req", _.req),
+    string.required[DefaultVariants]("reqDef", _.reqDef).addHints(smithy.api.Default(smithy4s.Document.fromString("default"))),
+    string.field[DefaultVariants]("optDef", _.optDef).addHints(smithy.api.Default(smithy4s.Document.fromString("default"))),
+    string.optional[DefaultVariants]("opt", _.opt),
```

Comment:
I'm not sure I fully understand why `field` is used here rather than `optional` and it is also used in other places where `required` was used prior. Is `field` only used when a default value is present and the field is not required?

Reply:
>  Is field only used when a default value is present and the field is not required 

Pretty much. 

* `optional` forces the type of the getter to be `S => Option[A]`
* `required` lets you do `S => A`, but automatically enriches the hints with `smithy.api.Required`
* `field` lets you do `S => A`, and doesn't enrich the hints with anything

Reply:
Got it yeah that makes sense